### ISSUE TITLE
xfce4-terminal: no sixel support

### ISSUE THREAD

**Issue by gepbird** [NixOS/nixpkgs#252845]
### Describe the bug
According to the [xfce4-terminal v1.1.0 changelogs](https://gitlab.xfce.org/apps/xfce4-terminal/-/blob/xfce4-terminal-1.1.0/NEWS#L39) ,sixel is supported, but it doesn't work.

### Steps To Reproduce
1. Install xfce.xfce4-terminal  and lsix
2. Run xfce4-terminal
3. Inside the terminal run lsix

### Expected behavior
Lsix should say sixel is supported.

### Screenshots with actual result

![](https://i.imgur.com/60pdSIo.png)

### Additional context
Maybe vte should be built with an additional flag, `-Dsixel=true` to fix this issue?

### Notify maintainers

vte: @bobby285271 @romildo @muscaln 
xfce: @astsmtl @antono 

### Metadata
```console
[user@system:~]$ nix-shell -p nix-info --run "nix-info -m"
 - system: `"x86_64-linux"`
 - host os: `Linux 6.1.49, NixOS, 23.11 (Tapir), 23.11pre520402.e7f38be3775b`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.17.0`
 - channels(root): `"nixos, stable-23.05"`
 - channels(gep): `"nixpkgs"`
 - nixpkgs: `/home/gep/.nix-defexpr/channels/nixpkgs`
```

**Comment by bobby285271** [comment]
This is not supported in stable vte release - https://github.com/GNOME/vte/commit/e0720853f59ebf54e022b4cd164de2d9b8a21fc1

**Comment by linsui** [comment]
vet unstable is also needed by blackbox. We can use the same vte unstable. https://github.com/NixOS/nixpkgs/pull/248682

**Comment by bobby285271** [comment]
There is https://github.com/NixOS/nixpkgs/pull/224104 which is rejected previously, I personally don't use the feature so I can't comment much here 😂

**Comment by linsui** [comment]
Yes, I put it in my overlays and it works well. Given that this feature is already used in terminals, I thought it's stable enough. @jtojnar Could you please reconsider the decision?

**Comment by jtojnar** [comment]
@linsui I am not aware of anything changing with regards to https://github.com/NixOS/nixpkgs/pull/224104#issuecomment-1492426531

**Comment by linsui** [comment]
@jtojnar What's your preferred methed to support features supported by upstream build in nixpkgs build? Is local override good to you?

**Comment by jtojnar** [comment]
I prefer `enableFoo` boolean input argument without an explicit attribute but note that the sixel feature is [**not supported** by upstream](https://gitlab.gnome.org/GNOME/vte/-/commit/e0720853f59ebf54e022b4cd164de2d9b8a21fc1).

**Comment by linsui** [comment]
Do you mean that we should add a `enableSixel` arg to terminals and use a overrided vte with sixel support when the arg is true?

> note that the sixel feature is [not supported by upstream](https://gitlab.gnome.org/GNOME/vte/-/commit/e0720853f59ebf54e022b4cd164de2d9b8a21fc1).

The vte lib is not our upstream. The blackbox and xfce terminal devs are our upstream. The sixel feature is supported by the terminal packages.

**Comment by jtojnar** [comment]
If the terminal emulator claims to support sixel out of the box, I would just use overridden vte internally in the emulator derivation.

> The sixel feature is supported by the terminal packages.

Then they are building on unstable foundations.

**Comment by linsui** [comment]
> If the terminal emulator claims to support sixel out of the box, I would just use overridden vte internally in the emulator derivation.

Yes, that what I do for blackbox in https://github.com/NixOS/nixpkgs/pull/248682. Not sure about xfce terminal. Local override is good for one package but I thought it's better to use the same unstable vte everywhere.

**Comment by jtojnar** [comment]
Unless the development of the two projects is co-ordinated, each will probably be tested with a different vte commit so tracking them separately will better match the development model.

**Comment by j4james** [comment]
FYI, Sixel broke when they switched to GTK 4. See https://gitlab.gnome.org/GNOME/vte/-/issues/2717

So if you want a build of vte with working Sixel, I think you need a fork from some point earlier than https://github.com/GNOME/vte/commit/c557b2300c835f0d22b6647c9155efdf101d0c2e.
